### PR TITLE
Improve compatibility of the memory allocation helper

### PR DESCRIPTION
**Pull request by skozin**
Align free memory pointer to 32 bytes when allocating memory, as it's currently done by the Solidity compiler (and was done by the `BytesLib` dependency that we're not using anymore).

* [x] Tests

### CODE REVIEW

File: contracts/common/lib/MemUtils.sol [hunk 1]
```diff
@@ -15,7 +15,9 @@ library MemUtils {
         assembly {
             result := mload(0x40)
             mstore(result, _len)
-            mstore(0x40, add(add(result, _len), 32))
+            let freeMemPtr := add(add(result, 32), _len)
+            // align free mem ptr to 32 bytes as the compiler does now
+            mstore(0x40, and(add(freeMemPtr, 31), not(31)))
```
Comment (TheDZhon) [inline contracts/common/lib/MemUtils.sol:20]: :+1: 

Got a flashback:
```cpp
#ifndef align_up
#define align_up(num, align) \
    (((num) + ((align) - 1)) & ~((align) - 1))
#endif
```

Comment (skozin) [reply contracts/common/lib/MemUtils.sol:20]: love the macro parentheses craziness

File: contracts/common/test_helpers/Assertions.sol [hunk 2]
```diff
@@ -0,0 +1,214 @@
+// SPDX-FileCopyrightText: 2023 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+/* See contracts/COMPILERS.md */
+pragma solidity 0.8.9;
```
Comment (TheDZhon) [inline contracts/common/test_helpers/Assertions.sol:5]: not sure it should be placed in the `common` though

**Review by TheDZhon — APPROVED** [review]
:heart:

File: contracts/common/test_helpers/Assertions.sol (hunk 2 above)
Comment (skozin) [reply contracts/common/test_helpers/Assertions.sol:5]: yeah the tests are using 0.8.9 and that's the problem. however, the `MemUtils` itself supports 0.4.24

Comment (skozin) [reply contracts/common/test_helpers/Assertions.sol:5]: not sure what to do here. should we move the tests to `0.8.9` given that the tested code is in `common`?

Comment (skozin) [reply contracts/common/test_helpers/Assertions.sol:5]: writing assertions supporting 0.4.24 would be a pain

Comment (TheDZhon) [reply contracts/common/test_helpers/Assertions.sol:5]: I guess we can proceed as is, too minorish for me

**Pull request merged**